pkg/querier/tripperware/queryrange/query_range_middlewares.go

```diff
@@ -89,8 +95,16 @@ func Middlewares(
 		queryRangeMiddleware = append(queryRangeMiddleware, tripperware.InstrumentMiddleware("step_align", metrics), StepAlignMiddleware)
 	}
 	if cfg.SplitQueriesByInterval != 0 {
-		staticIntervalFn := func(_ tripperware.Request) time.Duration { return cfg.SplitQueriesByInterval }
-		queryRangeMiddleware = append(queryRangeMiddleware, tripperware.InstrumentMiddleware("split_by_interval", metrics), SplitByIntervalMiddleware(staticIntervalFn, limits, prometheusCodec, registerer))
+		intervalFn := func(_ tripperware.Request) time.Duration { return cfg.SplitQueriesByInterval }
+		if cfg.SplitQueriesByIntervalMaxSplits != 0 {
```

Comment:
Shouldn't the limit be applied to both range splits and vertical spits?

https://github.com/cortexproject/cortex/blob/8a46d206db2e098f0df2dc2ba9d48c08f951f51f/pkg/querier/tripperware/shard_by.go#L40

Reply:
Technically this sets a limit for the total range and vertical splits for a given query. The number of vertical shards is static, so the max number of of splits for a given query becomes `split_queries_by_interval_max_splits` x `query_vertical_shard_size`. Because of this adding a separate limit for vertical sharding when the number of vertical shards is a static config would be redundant.